interpreters/python/Makefile

```diff
@@ -0,0 +1,165 @@
+CPYTHON_ZIP = v$(CPYTHON_VERSION).zip
+
+CPYTHON_UNPACKNAME = Python
+UNPACK ?= unzip -q -o
+
+MACHDEP=nuttx
+CONFIG_SITE=${CURDIR}/config.site
+CPYTHON_PATH=$(CURDIR)/$(CPYTHON_UNPACKNAME)
+
+BUILDIR=$(CURDIR)/build
+INSTALLDIR=$(CURDIR)/install
+HOSTBUILD=$(BUILDIR)/host
+HOSTINSTALL=$(INSTALLDIR)/host
+HOSTPYTHON=$(HOSTINSTALL)/bin/python3
+TARGETBUILD=$(BUILDIR)/target
+TARGETINSTALL=$(INSTALLDIR)/target
+TARGETLIBPYTHON=$(TARGETINSTALL)/libpython3.13.a
+TARGETMODULESPACK=$(TARGETBUILD)/lib/python313.zip
+TARGETMODULES=$(TARGETINSTALL)/lib/
+
+DEPPATH += --dep-path $(CPYTHON_UNPACKNAME)$(DELIM)Programs
+VPATH   += :$(CPYTHON_UNPACKNAME)$(DELIM)Programs
+
+$(CPYTHON_ZIP):
+	@echo "Downloading: $(CPYTHON_URL)/$(CPYTHON_ZIP)"
+	$(Q) $(call DOWNLOAD,$(CPYTHON_URL),$(CPYTHON_ZIP))
+
+$(CPYTHON_UNPACKNAME): $(CPYTHON_ZIP)
+	@echo "Unpacking: $(CPYTHON_ZIP) -> $(CPYTHON_UNPACKNAME)"
+	$(Q) $(UNPACK) $(CPYTHON_ZIP)
+	$(Q) mv	cpython-$(CPYTHON_VERSION) $(CPYTHON_UNPACKNAME)
+	@echo "Patching $(CPYTHON_UNPACKNAME)"
+	$(Q) patch -p1 -d $(CPYTHON_UNPACKNAME) < patch$(DELIM)0001-workaround-newlib-resource.h-limitations.patch
+	$(Q) patch -p1 -d $(CPYTHON_UNPACKNAME) < patch$(DELIM)0002-fix-various-uint32_t-unsigned-int-type-mismatch-issu.patch
+	$(Q) patch -p1 -d $(CPYTHON_UNPACKNAME) < patch$(DELIM)0003-reuse-wasm_assets.py-for-generating-an-archive-of-py.patch
+	$(Q) patch -p1 -d $(CPYTHON_UNPACKNAME) < patch$(DELIM)0004-recognize-nuttx-as-a-supported-OS.patch
+	$(Q) patch -p1 -d $(CPYTHON_UNPACKNAME) < patch$(DELIM)0005-gh-122907-Fix-Builds-Without-HAVE_DYNAMIC_LOADING-Se.patch
+	$(Q) patch -p1 -d $(CPYTHON_UNPACKNAME) < patch$(DELIM)0006-change-var-name-to-avoid-conflict-with-nuttx-unused_.patch
+	$(Q) patch -p1 -d $(CPYTHON_UNPACKNAME) < patch$(DELIM)0007-undef-atexit_register.patch
+	$(Q) patch -p1 -d $(CPYTHON_UNPACKNAME) < patch$(DELIM)0008-declare-struct-timeval.patch
+	$(Q) patch -p1 -d $(CPYTHON_UNPACKNAME) < patch$(DELIM)0009-include-nuttx-sys-select-header-to-define-FD_SETSIZE.patch
+	$(Q) patch -p1 -d $(CPYTHON_UNPACKNAME) < patch$(DELIM)0010-check-for-the-d_ino-member-of-the-structure-dirent.patch
+
+$(HOSTPYTHON):
+	mkdir -p $(HOSTBUILD)
+	mkdir -p $(HOSTINSTALL)
+	$(Q) ( \
+			cd $(HOSTBUILD) && $(CPYTHON_PATH)/configure \
+			 --with-pydebug \
+			 --prefix=$(HOSTINSTALL) \
+		 )
+	$(MAKE) -C $(HOSTBUILD) install
+
+$(TARGETBUILD)/Makefile: $(HOSTPYTHON)
+	$(Q) mkdir -p $(TARGETBUILD)/Modules
+	$(Q) mkdir -p $(TARGETMODULES)/python3.13
+	$(Q) ( cp Setup.local $(TARGETBUILD)/Modules/Setup.local )
+	$(Q) ( ls -la $(TARGETBUILD)/Modules/Setup.local )
+	$(Q) ( \
+			cd $(TARGETBUILD); \
+			CFLAGS="$(CFLAGS)"; \
+			ARCH=$(CONFIG_ARCH); \
+			ARCH_CHIP=$(CONFIG_ARCH_CHIP); \
+			echo "ARCH: $${ARCH}"; \
+			echo "ARCH_CHIP: $${ARCH_CHIP}"; \
+			ARCH="$${ARCH//-/}"; \
+			ARCH_CHIP="$${ARCH_CHIP//-/}"; \
+			echo "ARCH: $${ARCH}"; \
+			echo "ARCH_CHIP: $${ARCH_CHIP}"; \
+			CFLAGS="$$(echo "$${CFLAGS}" | sed 's/-Os //')" \
```

Comment:
whh meed

Reply:
Because we use NuttX's `CFLAGS` to define it, but Python's `configure` script defines the optimization by setting `OPT`. Keeping it on `CFLAGS` generates a conflitct.

Reply:
I recommend to put a comment here:

Something like:

```
Remove optimization flags from NuttX's CFLAGS because Python's configure script requires them in OPT. Having the flags in both places causes a conflict.
```